src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -90,5 +121,100 @@ public override int Execute()
+                var installer = _workloadInstaller.GetPackInstaller();
+                IEnumerable<PackInfo> workloadPackToUpdate = new List<PackInfo>();
+
+                TransactionalAction.Run(
+                    action: () =>
+                    {
+                        foreach (var manifest in manifestsToUpdate)
+                        {
+                            _workloadInstaller.InstallWorkloadManifest(manifest.manifestId, manifest.newVersion, sdkFeatureBand);
+                        }
+
+                        _workloadResolver.RefreshWorkloadManifests();
+
+                        workloadPackToUpdate = workloadIds
+                            .SelectMany(workloadId => _workloadResolver.GetPacksInWorkload(workloadId.ToString()))
+                            .Distinct()
+                            .Select(packId => _workloadResolver.TryGetPackInfo(packId));
+
+                        foreach (var packId in workloadPackToUpdate)
+                        {
+                            installer.InstallWorkloadPack(packId, sdkFeatureBand);
+                        }
+
+                        foreach (var workloadId in workloadIds)
+                        {
+                            _workloadInstaller.GetWorkloadInstallationRecordRepository()
+                                .WriteWorkloadInstallationRecord(workloadId, sdkFeatureBand);
+                        }
+                    },
+                    rollback: () => {
+                        try
+                        {
+                            _reporter.WriteLine(LocalizableStrings.RollingBackInstall);
+
+                            foreach (var manifest in manifestsToUpdate)
+                            {
+                                _workloadInstaller.InstallWorkloadManifest(manifest.manifestId, manifest.existingVersion, sdkFeatureBand);
+                            }
+
+                            foreach (var packId in workloadPackToUpdate)
+                            {
+                                installer.RollBackWorkloadPackInstall(packId, sdkFeatureBand);
+                            }
+
+                            foreach (var workloadId in workloadIds)
+                            {
+                                _workloadInstaller.GetWorkloadInstallationRecordRepository()
+                                    .DeleteWorkloadInstallationRecord(workloadId, sdkFeatureBand);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            // Don't hide the original error if roll back fails
+                            _reporter.WriteLine(string.Format(LocalizableStrings.RollBackFailedMessage, e.Message));
+                        }
+                    });
+            }
+            else
+            {
+                var installer = _workloadInstaller.GetWorkloadInstaller();
+                foreach (var workloadId in workloadIds)
```

Comment:
should have a test on when packs are not installed, this update is considered an install

Reply:
I think [this existing test ](https://github.com/dotnet/sdk/pull/17642/files#diff-7e506f78957029dd1bd1c2929f3e64a542fe4c0c29bbbbc54f95e29b5dc99499R40)covers this case, since right now the top level update command doesn't know if the pack is installed or not and it just asks the managed installer to install them all, knowing it will be a no op if the pack is already installed.